onnxruntime/core/providers/nuphar/scripts/symbolic_shape_infer.py

```diff
@@ -850,15 +852,30 @@ def _infer_Reshape(self, node):
 
             assert new_sympy_shape.count(-1) < 2
             if -1 in new_sympy_shape:
-                new_dim = total // non_deferred_size
+                # handle a special case where input_shape is [] and shape_value is [-1]
+                if len(input_shape) == 0 and len(shape_value) == 1:
+                    if node.output[0] in self.sympy_data_:
+                        sympy_shape = self.sympy_data_[node.output[0]][0]
+                    else:
+                        # create the sympy data if not exist
+                        self._pass_on_sympy_data(node)
+                        skip_pass_on_sympy_data = True
+                        if node.output[0] in self.sympy_data_:
```

Comment:
>node.output[0] in self.sympy_data [](start = 27, length = 33)

seems dead branch

Reply:
It's not dead branch. ```self._pass_on_sympa_data``` will add ```node.output[0]``` to ```self.sympy_data_``` in this case.